[Bug] Cannot upload JMC Agent probe templates after a failure

```
Jun 20, 2023 8:32:28 PM io.cryostat.core.log.Logger error
SEVERE: new_file.xml
Jun 20, 2023 8:32:28 PM io.cryostat.core.log.Logger error
SEVERE: HTTP 500: new_file.xml
io.vertx.ext.web.handler.HttpException: Internal Server Error
Caused by: io.cryostat.net.web.http.api.v2.ApiException: new_file.xml
	at io.cryostat.net.web.http.api.v2.ProbeTemplateUploadHandler.handle(ProbeTemplateUploadHandler.java:161)
	at io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler.handle(AbstractV2RequestHandler.java:108)
	at io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler.handle(AbstractV2RequestHandler.java:71)
	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
	at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.nio.file.FileAlreadyExistsException: new_file.xml
	at io.cryostat.core.agent.LocalProbeTemplateService.addTemplate(LocalProbeTemplateService.java:98)
	at io.cryostat.net.web.http.api.v2.ProbeTemplateUploadHandler.handle(ProbeTemplateUploadHandler.java:136)
	... 11 more
Jun 20, 2023 8:32:28 PM org.slf4j.impl.JDK14LoggerAdapter fillCallerData
SEVERE: 10.89.1.31 - - [Tue, 20 Jun 2023 20:32:28 GMT] 11ms "POST /api/v2/probes/jmc-agent-probe.xml HTTP/1.1" 500 96 bytes "http://localhost:8181/events?agentTab=agent-template" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0"
```

See #238 . After failing to upload a template once, a retry with the file corrected fails with the above. Once in this state it doesn't appear to be recoverable - all upload attempts will fail this way.

--- a/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
+++ b/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
@@ -87,18 +87,19 @@ public class LocalProbeTemplateService implements ProbeTemplateService {
         }
     }
 
-    public void addTemplate(InputStream inputStream, String filename)
+    public ProbeTemplate addTemplate(InputStream inputStream, String filename)
             throws FileAlreadyExistsException, IOException, SAXException {
+        Path path = fs.pathOf(env.getEnv(TEMPLATE_PATH), filename);
+        if (fs.exists(path)) {
+            throw new FileAlreadyExistsException(filename);
+        }
         try (inputStream) {
             ProbeTemplate template = new ProbeTemplate();
             template.setFileName(filename);
             // If validation fails this will throw a ProbeValidationException with details
             template.deserialize(inputStream);
-            Path path = fs.pathOf(env.getEnv(TEMPLATE_PATH), filename);
-            if (fs.exists(path)) {
-                throw new FileAlreadyExistsException(template.getFileName());
-            }
             fs.writeString(path, template.serialize());
+            return template;
         }
     }
 
@@ -138,8 +139,8 @@ public class LocalProbeTemplateService implements ProbeTemplateService {
                         Path fileName = path.getFileName();
                         if (fileName != null) {
                             ProbeTemplate template = new ProbeTemplate();
-                            template.deserialize(stream);
                             template.setFileName(fileName.toString());
+                            template.deserialize(stream);
                             templates.add(template);
                         }
                     }

--- a/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
+++ b/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
@@ -91,7 +91,8 @@ public class LocalProbeTemplateService implements ProbeTemplateService {
             throws FileAlreadyExistsException, IOException, SAXException {
         Path path = fs.pathOf(env.getEnv(TEMPLATE_PATH), filename);
         if (fs.exists(path)) {
-            throw new FileAlreadyExistsException(filename);
+            throw new FileAlreadyExistsException(
+                    String.format("Probe template \"%s\" already exists.", filename));
         }
         try (inputStream) {
             ProbeTemplate template = new ProbeTemplate();

--- a/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
+++ b/src/main/java/io/cryostat/core/agent/LocalProbeTemplateService.java
@@ -91,6 +91,7 @@ public class LocalProbeTemplateService implements ProbeTemplateService {
             throws FileAlreadyExistsException, IOException, SAXException {
         try (inputStream) {
             ProbeTemplate template = new ProbeTemplate();
+            template.setFileName(filename);
             // If validation fails this will throw a ProbeValidationException with details
             template.deserialize(inputStream);
             Path path = fs.pathOf(env.getEnv(TEMPLATE_PATH), filename);
@@ -108,9 +109,10 @@ public class LocalProbeTemplateService implements ProbeTemplateService {
         }
     }
 
-    public String getTemplate(String templateName) throws IOException, SAXException {
+    public String getTemplateContent(String templateName) throws IOException, SAXException {
         Path probeTemplatePath = fs.pathOf(env.getEnv(TEMPLATE_PATH), templateName);
         ProbeTemplate template = new ProbeTemplate();
+        template.setFileName(templateName);
         template.deserialize(fs.newInputStream(probeTemplatePath));
         return template.serialize();
     }

--- a/src/main/java/io/cryostat/core/agent/ProbeTemplate.java
+++ b/src/main/java/io/cryostat/core/agent/ProbeTemplate.java
@@ -107,7 +107,7 @@ public class ProbeTemplate {
         }
 
         Document document = builder.parse(stream);
-        stream.close();
+        stream.trulyClose();
         NodeList elements;
 
         // parse global configurations

